File: Day-3/Two City Scheduling.py
"""
Name of the problem :- Two City Scheduling
"""

class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        res = sorted(costs, key = lambda sub: abs(sub[1] - sub[0]), reverse=True) 
        a = 0
        b = 0
        cost = 0
        val = len(res)
        N = val//2
        
        for i in range(val):
            if res[i][0]<res[i][1] and a<N:
                a = a+1
                cost = cost+res[i][0]
                
            elif res[i][0]>=res[i][1] and b<N:
                b = b+1
                cost = cost+res[i][1]
                
            elif res[i][0]<res[i][1] and a==N:
                b = b+1
                cost = cost+res[i][1]
                
            elif res[i][0]>=res[i][1] and b==N:
                a = a+1
                cost = cost+res[i][0]
                
        return cost

File: Day-3/Two City Scheduling.cpp
static bool cmp(vector <int> a, vector <int> b)
{
      return abs(a[0] - a[1]) > abs(b[0] - b[1]);
}
class Solution {
public:
    int twoCitySchedCost(vector<vector<int>>& costs) 
    {
        sort(costs.begin(), costs.end(), cmp);   
        long long int i, a, b, n, sum=0;
        n=costs.size();
        a=b=(n/2);
        for(i=0;i<n;i++)
        {
            if((costs[i][0]<=costs[i][1] && b>0) || a==0)
            {    
                b--;
                sum+=costs[i][0];
            }
            else
            {
                a--;
                sum+=costs[i][1];
            }
        
        }
        return sum;
    }
};